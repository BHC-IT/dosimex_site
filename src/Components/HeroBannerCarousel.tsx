import * as React from 'react';
import * as CSS from 'csstype';
import Image from 'next/image';

import { useIsMobile } from '../Hooks/useIsMobile';


interface IStyles {
	[key: string]: CSS.Properties
}

interface ISlideProps {
	name: string,
	text: string,
	vw: number,
}

interface ITextProps {
	text: string[],
	vw: number,
}

const images = [
	"ScreenCarousel1.png",
	"ScreenCarousel2.png",
	"ScreenCarousel3.png",
	"ScreenCarousel4.png",
]


const Slide = ({name, text, vw}: ISlideProps) =>
	<div style={{width: "25%", float: "left"}}>
			<Image
				src={`/Images/${name}`}
				width={29*vw}
				height={16.32*vw}
				layout={"fixed"}
			/>
		<p style={{fontStyle: "italic", float: "right", marginTop: "7vh",padding: "0 10px", fontSize: "1.8rem", color: "var(--grey)"}}>{text}</p>
	</div>


const SlideWrapper = ({text, vw}: ITextProps) => {
	return (
		<div style={{overflow: "hidden"}}>
			<div style={{position: "relative", width: "400%", margin: 0, left: 0, animation: "10s slidy infinite" }}>
				{images.map((e, i) => <Slide name={e} text={text[i]} key={e} vw={vw} />)}
			</div>
		</div>
	)
}

const HeroBannerCarousel = ({text}: any) => {

	const [vw, setvw] = React.useState<number>(1500);

	React.useEffect(() => {
		setvw(window.innerWidth / 100)
	}, [])

	console.log(vw);
	const style = useIsMobile(styles);

	if (style === null)
		return <></>

	return (
		<div style={{}}>
			<div style={{ width: `${30 * vw}px`, height: `${19 * vw}px` }} >
				<Image
					src="/Images/MockupCarousel.png"
					layout={"fill"}
				/>
			</div>
			<div style={{position: "absolute", left: `${0.5 * vw}px`, top: `${0.7 * vw}px`}}>
				<SlideWrapper text={text} vw={vw} />
			</div>
		</div>
	)

}

export default HeroBannerCarousel;

export const styles = (mobile: boolean): IStyles => ({

})