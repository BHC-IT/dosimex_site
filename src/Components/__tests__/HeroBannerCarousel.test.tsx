import { render } from '@testing-library/react'
import { describe, it, expect, vi } from 'vitest'

import HeroBannerCarousel, { styles } from '../HeroBannerCarousel'

// Mock useIsMobile hook
vi.mock('../../Hooks/useIsMobile', () => ({
	useIsMobile: vi.fn(() => styles()),
}))

// Mock window.innerWidth
Object.defineProperty(window, 'innerWidth', {
	writable: true,
	configurable: true,
	value: 1024,
})

describe('HeroBannerCarousel Component', () => {
	const mockText = ['Text 1', 'Text 2', 'Text 3', 'Text 4']

	it('should render with text array', () => {
		const { container } = render(<HeroBannerCarousel text={mockText} />)
		expect(container).toBeInTheDocument()
	})

	it('should render without text (fixed with TypeScript improvements)', () => {
		// After TypeScript fixes, this should handle undefined text gracefully
		const { container } = render(<HeroBannerCarousel />)
		expect(container).toBeInTheDocument()
	})

	it('should match snapshot with text', () => {
		const { container } = render(<HeroBannerCarousel text={mockText} />)
		expect(container.firstChild).toMatchSnapshot()
	})

	it('should handle undefined text after TypeScript fixes', () => {
		// After TypeScript fixes, this should handle undefined text gracefully
		const { container } = render(<HeroBannerCarousel text={undefined} />)
		expect(container).toBeInTheDocument()
	})

	it('should handle different text lengths', () => {
		const shortText = ['Text 1', 'Text 2']
		const { container } = render(<HeroBannerCarousel text={shortText} />)
		expect(container).toBeInTheDocument()
	})
})
