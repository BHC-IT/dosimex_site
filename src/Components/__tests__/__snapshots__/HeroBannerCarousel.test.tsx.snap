// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HeroBannerCarousel Component > Rendering > should render with text array 1`] = `
<div>
  <div
    style="width: 307.2px; height: 194.56px; position: relative;"
  >
    <img
      alt="Main carousel interface"
      data-nimg="1"
      decoding="async"
      height="194.56"
      src="/_next/image?url=%2FImages%2FMockupCarousel.png&w=640&q=85"
      srcset="/_next/image?url=%2FImages%2FMockupCarousel.png&w=384&q=85 1x, /_next/image?url=%2FImages%2FMockupCarousel.png&w=640&q=85 2x"
      style="color: transparent;"
      width="307.2"
    />
  </div>
  <div
    style="position: absolute; left: 5.12px; top: 7.167999999999999px;"
  >
    <div
      style="overflow: hidden;"
    >
      <div
        style="position: relative; width: 400%; margin: 0px; left: 0px; animation: 10s slidy infinite;"
      >
        <div
          style="width: 25%; float: left;"
        >
          <img
            alt="Screenshot ScreenCarousel1.png"
            data-nimg="1"
            decoding="async"
            height="167.1168"
            loading="lazy"
            src="/_next/image?url=%2FImages%2FScreenCarousel1.png&w=640&q=80"
            srcset="/_next/image?url=%2FImages%2FScreenCarousel1.png&w=384&q=80 1x, /_next/image?url=%2FImages%2FScreenCarousel1.png&w=640&q=80 2x"
            style="color: transparent;"
            width="296.96"
          />
          <p
            style="font-style: italic; float: right; margin-top: 7vh; padding: 0px 10px; font-size: 1.8rem; color: var(--grey);"
          >
            Advanced radiation dose management system
          </p>
        </div>
        <div
          style="width: 25%; float: left;"
        >
          <img
            alt="Screenshot ScreenCarousel2.png"
            data-nimg="1"
            decoding="async"
            height="167.1168"
            loading="lazy"
            src="/_next/image?url=%2FImages%2FScreenCarousel2.png&w=640&q=80"
            srcset="/_next/image?url=%2FImages%2FScreenCarousel2.png&w=384&q=80 1x, /_next/image?url=%2FImages%2FScreenCarousel2.png&w=640&q=80 2x"
            style="color: transparent;"
            width="296.96"
          />
          <p
            style="font-style: italic; float: right; margin-top: 7vh; padding: 0px 10px; font-size: 1.8rem; color: var(--grey);"
          >
            Real-time monitoring and analysis tools
          </p>
        </div>
        <div
          style="width: 25%; float: left;"
        >
          <img
            alt="Screenshot ScreenCarousel3.png"
            data-nimg="1"
            decoding="async"
            height="167.1168"
            loading="lazy"
            src="/_next/image?url=%2FImages%2FScreenCarousel3.png&w=640&q=80"
            srcset="/_next/image?url=%2FImages%2FScreenCarousel3.png&w=384&q=80 1x, /_next/image?url=%2FImages%2FScreenCarousel3.png&w=640&q=80 2x"
            style="color: transparent;"
            width="296.96"
          />
          <p
            style="font-style: italic; float: right; margin-top: 7vh; padding: 0px 10px; font-size: 1.8rem; color: var(--grey);"
          >
            Comprehensive reporting and compliance features
          </p>
        </div>
        <div
          style="width: 25%; float: left;"
        >
          <img
            alt="Screenshot ScreenCarousel4.png"
            data-nimg="1"
            decoding="async"
            height="167.1168"
            loading="lazy"
            src="/_next/image?url=%2FImages%2FScreenCarousel4.png&w=640&q=80"
            srcset="/_next/image?url=%2FImages%2FScreenCarousel4.png&w=384&q=80 1x, /_next/image?url=%2FImages%2FScreenCarousel4.png&w=640&q=80 2x"
            style="color: transparent;"
            width="296.96"
          />
          <p
            style="font-style: italic; float: right; margin-top: 7vh; padding: 0px 10px; font-size: 1.8rem; color: var(--grey);"
          >
            User-friendly interface for efficient workflow
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
